Remove Global Installation  References from Unit 1 (#208)
Ada-Developers-Academy/core-unit-1: intro-to-tests/intro-to-pytest.md

[`pytest`](https://docs.pytest.org/en/stable/) is one of many testing tools for the Python language.

> 
> 
> Before we install `pytest`, let's recall the steps we took in Precourse the commands to install Python packages. First, we need to navigate to the directory that we want to setup our virtual environment in. We can use the `pwd` command to check if we are currently inside of the desired directory, if not then we can use the `cd` command to navigate to the correct one. After we are inside the correct directory we can the following commands to create and then activate our virtual environment:

Reviewer: ```suggestion
Before we install `pytest`, let's recall the steps we took in Precourse the commands to install Python packages:
1. First, we need to navigate to the directory that we want to setup our virtual environment in. 
    - We can use the `pwd` command to check if we are currently inside of the desired directory. 
    - If we're not in the right folder, then we can use the `cd` command to navigate to the correct one. 
2. After we are inside the correct directory we can use the following commands to create and then activate our virtual environment:
```

```bash
$ python3 -m venv venv
$ source venv/bin/activate
```

Remember that the first command creates a virtual environment under the name `venv` and the second command activates our virtual environment. After running the activation command, we should now see that each line of our terminal now begins with `(venv)`. This lets us know that our virtual environment has been activated. 

### !callout-info

## Virtual Environments

> At high level, virtual environments are used to  install and manage packages separately from other projects that could be using the same packages. For instance, if project A requires an older version of `pytest` than project B, using a virtual environment allows both projects to exist on the same machine without have to upgrade/downgrade `pytest` to meet the conflicting version requirements. We will discuss virtual environments in greater detail later in `Managing Packages` and when discussing the `Adagrams` project.

Reviewer: ```suggestion
At high level, virtual environments are used to install and manage packages separately from other projects that could be using the same packages. For instance, if project A requires an older version of `pytest` than project B, using a virtual environment allows both projects to exist on the same machine without needing to upgrade/downgrade `pytest` to meet the conflicting version requirements. We will discuss virtual environments in greater detail later in `Managing Packages` and when discussing the `Adagrams` project.
```

### !end-callout

Now let's use the command below to install `pytest`:

```bash
(venv) pip3 install -U pytest
```

> ### !callout-danger

Reviewer: To me, this doesn't feel like it fits a `callout-danger` since it's reminding folks about something they've already seen. Would you be able to swap this over to an info callout while in the file?
```suggestion
### !callout-info
```

Author: Sorry, could you explain what you mean by "...while in the file"? I want to make sure I'm not missing anything.

Reviewer: All good! I was checking if you were comfortable to make this update while you're already making changes to this lesson, even though this particular change is outside the scope of the global installation references work you were doing.

Author: Oh! Yes, that's fine by me!

Take care to notice the following:

1. In our project directory, there are 0 files, and 3 folders: `projectname`, `tests`, and `venv`.
   - We will replace the folder name `projectname` with our project's name
2. In our `projectname` folder, there is a file named `somefile.py`
   - This represents any Python files that include the functions we're writing
3. In our `tests` folder, there is a file named `test_somefile.py`
   - This represents a test file that is responsible for testing `projectname/somefile.py`

> 1. There are some other files such as `__init__.py`

Reviewer: Since the others bullets moved to sequential numbers, I think we need to update the number here as well for the order to flow right.
```suggestion
4. There are some other files such as `__init__.py`
```

Author: Oops! Let me fix that!

   - This represents a test file that is responsible for testing `projectname/somefile.py`

> 1. There are some other files: `__init__.py` and `context.py`

Reviewer: GitHub won't let me select the first bullet on line 55 to suggest a change, but we should update it to acknowledge there's also a `venv` folder as well.

```suggestion
1. There are some other files such as `__init__.py`
```
